Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedListItemView.swift

```diff
@@ -48,12 +62,23 @@ public struct PreplannedListItemView: View {
     }
     
     @ViewBuilder private var downloadButton: some View {
-        Button {
-            
-        } label: {
-            Image(systemName: "arrow.down.circle")
+        switch model.status {
+        case .downloaded:
+            Image(systemName: "checkmark.circle")
+                .foregroundStyle(.secondary)
+        case .downloading:
+            if let job = model.job {
+                ProgressView(job.progress)
+                    .progressViewStyle(.gauge)
+            }
+        default:
+            Button {
+                mapViewModel.selectedMap = OfflineMapAreasView.MapViewModel.SelectedMap.preplannedMap(model)
```

Comment:
```swift
                mapViewModel.selectedMap = .preplannedMap(model)
```
It can be shortened; see my comment above on not using the enum at all.